expfmt/text_parse.go

```diff
@@ -574,7 +574,14 @@ func (p *TextParser) readingType() stateFn {
 	if p.readTokenUntilNewline(false); p.err != nil {
 		return nil // Unexpected end of input.
 	}
-	metricType, ok := dto.MetricType_value[strings.ToUpper(p.currentToken.String())]
+	typString := strings.ToUpper(p.currentToken.String())
+	// OpenMetrics uses UNKNOWN for untyped metrics instead of UNTYPED, so we add
```

Comment:
Hm, this PR was erroring when somebody tries to use OM with text parse, but we still try to handle OM here? 🤔 What am I missing? Also a candidate for a separate PR that enables OM decoding, if we want

Reply:
This was a bit of a stupid hack to try to return a better error message when someone does try to do this.  I do agree it's in conflict with refusing to decode openmetrics